src/main/java/org/reactome/qa/doisuggester/ReactionlikeEvent.java

```diff
@@ -0,0 +1,292 @@
+	{
+		List<GKInstance> newInstanceEdits = new ArrayList<>();
+		// A naive approach to this problem would simply look at list sizes, but that wont' work...
+		//
+		// Normally, you'd think that the list of InstanceEdits from the current list will *always* contain the
+		// same InstanceEdits as the list from the previous Release as well as any new ones - but Joel pointed out that sometimes InstanceEdits
+		// in previous versions have gotten lost due to bugs/data issues, so comparing the previous list to the current list
+		// could result in the previous list having *fewer* items. So, you need to examine the elements - you can not use list size alone.
+		//
+		// We need to go through ALL of the InstanceEdits in the current database's list and compare them to ALL of the InstanceEdits
+		// in the list from the previous database to determine if it's truly new.
+		for (GKInstance instanceEdit : currentInstanceEdits)
+		{
+			int i = 0;
+			boolean found = false;
+			while (!found && i < prevInstanceEdits.size())
+			{
+				found = instanceEdit.getDBID().equals(prevInstanceEdits.get(i).getDBID());
+				i++;
+			}
+			// if we went through ALL previous InstanceEdits and found == false, then instanceEdit is *new*!
+			if (!found)
+			{
+				newInstanceEdits.add(instanceEdit);
+			}
+		}
+		return newInstanceEdits;
+	}
+
+	/**
+	 * Gets the furthest ancestors of this Reaction-like event which does not have newInstanceEdit as one of its InstanceEdits.
+	 * This method will perform a recursive search from a reaction-like event up to a top-level pathway, searching for a pathway that
+	 * does not have a "new" InstanceEdit. A set of all such pathways is returned.
+	 * @param newInstanceEdit a "new" InstanceEdit.
+	 * @return The pathway objects.
+	 */
+	public Set<GKInstance> getFurthestAncestorsWithoutNewInstanceEdit(GKInstance newInstanceEdit)
+	{
+		List<GKInstance> parents;
+		try
+		{
+			this.ancestors.addAll(this.getFurthestAncestorWithoutNewInstanceEdit(this.getUnderlyingInstance(), newInstanceEdit));
+		}
+		catch (Exception e)
+		{
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return this.ancestors;
+	}
+
+	/**
+	 * Private recursive method to search up the Pathway hierarchy.
+	 * This method gets the ancestors of a pathway/RLE that is furthest up the tree and also does not have the same InstanceEdit.
+	 * @param A pathway to examine.
+	 * @param newInstanceEdit the InstanceEdit to check for.
+	 * @return The pathway objects.
+	 */
+	private Set<GKInstance> getFurthestAncestorWithoutNewInstanceEdit(GKInstance pathway, GKInstance newInstanceEdit)
+	{
+		try
+		{
+			List<GKInstance> parents = (List<GKInstance>) pathway.getReferers(ReactomeJavaConstants.hasEvent);
+			// Only proceed if there are valid parents. If there are no valid parents, then don't do anything because
+			// this process only makes sense if pathway has parents.
+			if (! (parents == null || parents.isEmpty()) )
+			{
```

Comment:
Should the pathway itself be checked?  If the pathway does not have the new instance edit and has also no parents, would it make sense to return the set containing just the pathway as the furthest without new instance edit?

Reply:
Also, if there are parents for the pathway, but the pathway is already known to not have the instance edit, the addition to the set can happen before/outside the pathway's parent loop and the check on line 267-278 can be replaced with just adding the parent pathway to the set when the instance edit isn't found.

Reply:
A pathway not have parents means that it's a top-level pathway. Or are there pathways with no parents that are also _not_ top-level pathways?

Reply:
Also, at this point in the program, I am pretty sure that `pathway` _must_ have the new InstanceEdit. If you reach this part of the program through the overload of this function, the input instance will have the new InstanceEdit and the whole point is to check the _ancestors_.  If you get to this point through this function's recursive call, it's because the new InstanceEdit was found on `pathway` when it was still `parent` in the call further up the stack.

Reply:
> Also, if there are parents for the pathway, but the pathway is already known to not have the instance edit, the addition to the set can happen before/outside the pathway's parent loop and the check on line 267-278 can be replaced with just adding the parent pathway to the set when the instance edit isn't found.

I'm a bit confused by what you mean here. I'm not checking to see if `pathway` has the new InstanceEdit or not, I'm checking the _ancestors_.

Reply:
There are po

> If a pathway does not have parents, that means that it's a top-level pathway. Or are there pathways with no parents that are also _not_ top-level pathways?

There are potentially pathways with no parents that are not top level pathways.  They would be considered orphan pathways.  This doesn't usually happen and is likely a pathway connection QA issue that they should have one or more parents especially if the pathway is released, but it is possible.

Reply:
Hmm. I don't think orphan pathways had been considered. I thought they were an error that needed to be corrected before being published. Still, I think this code will be OK with orphan pathways. If an orphan pathway is encountered, it should already have the new InstanceEdit at this point in the program and since there are no more ancestors and since this function is looking for unmodified ancestors, I think it makes sense to do nothing. What do you think?

Reply:
Yeah, that sounds right to me.